src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/OneDayFileJsonParser.java

```diff
@@ -0,0 +1,122 @@
+                            jsonMap.get("destination").getClass() != Long.class ||
+                            jsonMap.get("duration").getClass() != Integer.class ||
+                            !( "OK".equals(jsonMap.get("status_code")) || "KO".equals(jsonMap.get("status_code")) ) ||
+                            jsonMap.get("status_description").getClass() != String.class )
+                    {metricsInfoDataSet.setFieldsErrorsRowsCounter(metricsInfoDataSet.getFieldsErrorsRowsCounter() +1);}
+
+                    // relation between OK/KO calls
+                    if ( "OK".equals(jsonMap.get("status_code")) ) {okCallsCounter ++;}
+                    if ( "KO".equals(jsonMap.get("status_code")) ) {koCallsCounter ++;}
+
+                    // average call duration
+                    if ( jsonMap.get("duration").getClass() == Integer.class ) {
+                        metricsInfoDataSet.setAverageCallDuration(
+                                ( metricsInfoDataSet.getAverageCallDuration() * callsCounter + (int)jsonMap.get("duration") )
+                                / ++callsCounter );
+                    }
+
+                    // MSG type of file row
+                } else if ("MSG".equals( jsonMap.get("message_type") )) {
+
+                    // rows with missing fields
+                    if ("".equals(jsonMap.get("timestamp")) ||
+                            "".equals(jsonMap.get("origin")) ||
+                            "".equals(jsonMap.get("destination")) ||
+                            "".equals(jsonMap.get("message_status")) )
+                    {metricsInfoDataSet.setMissingFieldsRowsCounter(metricsInfoDataSet.getMissingFieldsRowsCounter() +1);}
+
+                    // messages with blank content
+                    if ("".equals(jsonMap.get("message_content")) ) {
+                        metricsInfoDataSet.setBlankContentMessagesCounter(metricsInfoDataSet.getBlankContentMessagesCounter() +1);}
+
+                    // rows with field errors
+                    if ( jsonMap.get("timestamp").getClass() != Long.class ||
+                            jsonMap.get("origin").getClass() != Long.class ||
+                            jsonMap.get("destination").getClass() != Long.class ||
+                            jsonMap.get("message_content").getClass() != String.class ||
+                            !( "DELIVERED".equals(jsonMap.get("message_status")) || "SEEN".equals(jsonMap.get("message_status")) ))
+                    {metricsInfoDataSet.setFieldsErrorsRowsCounter(metricsInfoDataSet.getFieldsErrorsRowsCounter() +1);}
+
+                } else if ("".equals( jsonMap.get("message_type") )){
+                    metricsInfoDataSet.setMissingFieldsRowsCounter(metricsInfoDataSet.getMissingFieldsRowsCounter() +1);
+                } else {
+                    // field error in message_type
+                    metricsInfoDataSet.setFieldsErrorsRowsCounter(metricsInfoDataSet.getFieldsErrorsRowsCounter() +1);
```

Comment:
Myslíš, že by se dala inkrementace metricsInfoData.fieldsErrorsRowCounter, nějak zjednodušit, než používat setter, který bere hodnotu z getteru, tu potom inkrementuje tady a poté ji vrací do setteru?

Reply:
předěláno ...